openwisp_controller/config/base/whois.py

```diff
@@ -0,0 +1,110 @@
+from ipaddress import ip_address, ip_network
+
+from django.core.cache import cache
+from django.core.exceptions import ValidationError
+from django.db import models, transaction
+from django.utils.translation import gettext_lazy as _
+from jsonfield import JSONField
+
+from openwisp_utils.base import TimeStampedEditableModel
+
+from ..whois.service import WhoIsService
+
+
+class AbstractWhoIsInfo(TimeStampedEditableModel):
+    """
+    Abstract model to store WhoIs information
+    for a device.
+    """
+
+    id = None
+    ip_address = models.GenericIPAddressField(db_index=True, primary_key=True)
+    isp = models.CharField(
+        max_length=100,
+        blank=True,
+        help_text=_("Organization associated with registered Autonomous System Number"),
+    )
+    asn = models.CharField(
+        max_length=6,
+        blank=True,
+        help_text=_("Autonomous System Number"),
```

Comment:
If I am not mistaken this is not even shown in the UI right? We can shorten it..
```suggestion
        help_text=_("ASN"),
```